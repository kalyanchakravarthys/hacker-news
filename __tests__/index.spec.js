import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Index from "../pages/index.js";

describe("Index Page", () => {
  it('renders fine"', () => {
    const app = shallow(<Index />);
    expect(app.find(".stories-page-no").text()).toEqual("Page No: ");
  });
});

describe("Index Snapshot Testing", () => {
  it('Index matches its snapshot"', () => {
    const component = renderer.create(<Index />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});