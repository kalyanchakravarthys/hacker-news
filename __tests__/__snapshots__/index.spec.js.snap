// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Index matches its snapshot" 1`] = `
<div>
  <div
    className="story-list"
  >
    <div
      className="stories-header"
    >
      <a
        href="https://news.ycombinator.com"
      >
        <img
          alt="Y"
          className="stories-source-logo"
          src="https://news.ycombinator.com/y18.gif"
        />
      </a>
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        top
      </a>
      |
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        new
      </a>
    </div>
  </div>
  <div>
    <a
      href="/?page=NaN"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      More
    </a>
    <div
      Style="float:right;"
      className="stories-page-no"
    >
      Page No: 
    </div>
  </div>
</div>
`;
